app/uk/gov/hmrc/tariffclassificationfrontend/controllers/ReviewCaseController.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tariffclassificationfrontend.controllers
+
+import javax.inject.{Inject, Singleton}
+import play.api.i18n.MessagesApi
+import play.api.mvc._
+import uk.gov.hmrc.tariffclassificationfrontend.config.AppConfig
+import uk.gov.hmrc.tariffclassificationfrontend.forms.ReviewForm
+import uk.gov.hmrc.tariffclassificationfrontend.models.ReviewStatus.ReviewStatus
+import uk.gov.hmrc.tariffclassificationfrontend.models.{Case, CaseStatus}
+import uk.gov.hmrc.tariffclassificationfrontend.service.CasesService
+import uk.gov.hmrc.tariffclassificationfrontend.views
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future.successful
+
+@Singleton
+class ReviewCaseController @Inject()(authenticatedAction: AuthenticatedAction,
+                                     override val caseService: CasesService,
+                                     override val messagesApi: MessagesApi,
+                                     override implicit val config: AppConfig) extends RenderCaseAction {
+
+  override protected def redirect: String => Call = routes.CaseController.trader
+
+  override protected def isValidCase: Case => Boolean = c => Set(CaseStatus.COMPLETED, CaseStatus.CANCELLED).contains(c.status)
```

Comment:
`Set(CaseStatus.COMPLETED, CaseStatus.CANCELLED)` can be saved in a private lazy val
There is no point on creating it from scratch all the times you call `isValidCase`

Reply:
see the other actions - we have two similar examples

Reply:
Seems a little like a premature optimisation. Personally I'd prefer to reduce the visual clutter caused by excessive fields, especially given its only used in one place.